xdem/spstats.py

```diff
@@ -1003,22 +1009,273 @@ def patches_method(dh : np.ndarray, mask: np.ndarray[bool], gsd : float, area_si
+                         edgecolor='white')
+    ax0.set_ylabel('Sample count')
+    # In case the axis value does not agree with the scale (e.g., 0 for log scale)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        ax0.set_ylim((0,1.1*np.max(list_counts)))
+        ax0.set_xlim((np.min(interval_var_1.left),np.max(interval_var_1.right)))
+    ax0.ticklabel_format(axis='y',style='sci',scilimits=(0,0))
+    ax0.spines['top'].set_visible(False)
+    ax0.spines['right'].set_visible(False)
+    # Try to identify if the count is always the same
+    if np.sum(~(np.abs(list_counts[0] - np.array(list_counts)) < 5)) <= 2:
+        ax0.text(0.5, 0.5, "Fixed number of\nsamples: " + '{:,}'.format(int(list_counts[0])), ha='center', va='center',
+                 fontweight='bold', transform=ax0.transAxes, bbox=dict(facecolor='white', alpha=0.8))
+
+    # Second, a vertical axis on the right to plot the sample histogram of the second variable
+    ax1 = fig.add_subplot(grid[3:, -3:])
+    ax1.set_yscale(scale_var_2)
+    ax1.set_yticklabels([])
+
+    # Plot the histogram manually with fill_between
+    interval_var_2 = pd.IntervalIndex(df_sub[var_name_2])
+    df_sub['var2_mid'] = interval_var_2.mid.values
+    unique_var_2 = np.unique(df_sub.var2_mid)
+    list_counts = []
+    for i in range(len(unique_var_2)):
+        df_var2 = df_sub[df_sub.var2_mid == unique_var_2[i]]
+        count = np.nansum(df_var2['count'].values)
+        list_counts.append(count)
+        ax1.fill_between([0, count], [df_var2[var_name_2].values[0].left] * 2, [df_var2[var_name_2].values[0].right] * 2, facecolor=plt.cm.Greys(0.75),
+                         alpha=1, edgecolor='white')
+    ax1.set_xlabel('Sample count')
+    # In case the axis value does not agree with the scale (e.g., 0 for log scale)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        ax1.set_xlim((0,1.1*np.max(list_counts)))
+        ax1.set_ylim((np.min(interval_var_2.left),np.max(interval_var_2.right)))
+    ax1.ticklabel_format(axis='x',style='sci',scilimits=(0,0))
+    ax1.spines['top'].set_visible(False)
+    ax1.spines['right'].set_visible(False)
+    # Try to identify if the count is always the same
+    if np.sum(~(np.abs(list_counts[0] - np.array(list_counts)) < 5)) <= 2:
+        ax1.text(0.5, 0.5, "Fixed number of\nsamples: " + '{:,}'.format(int(list_counts[0])), ha='center', va='center',
+                 fontweight='bold', transform=ax1.transAxes, rotation=90, bbox=dict(facecolor='white', alpha=0.8))
+
+    # Third, an axis to plot the data as a colored grid
+    ax = fig.add_subplot(grid[3:, :-3])
+
+    # Define limits of colormap is none are provided, robust max and min using percentiles
+    if vmin is None and vmax is None:
+        vmax = np.nanpercentile(df_sub[statistic_name].values, 99)
+        vmin = np.nanpercentile(df_sub[statistic_name].values, 1)
+
+    # Create custom colormap
```

Comment:
We discussed in the "terrain attribute" discussions that maybe we should have a set of colormaps somewhere in xdem. Should we raise an issue about that?

Reply:
Raised in #191